### PR TITLE
docs: document flaky test handling

### LINKED ISSUE #12584: Document guidelines on how to handle flaky tests
**Description**

Add a guide on what to do when a flaky test is encountered. The guide should enable contributors to make progress when flaky tests occur in their contributions.

This was an action derived from one of the ZPA team's recent retros.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add a guide on what to do when a flaky test is encountered. The guide is included in the ci.md section of our developer docs. This section is also linked inside our CONTRIBUTING.md guide to ensure that any contributors are aware of what to do in such an occurrence.

## Related issues

This was an action derived from one of the ZPA team's recent retros.

closes #12584

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by npepinpe — APPROVED** [review]
:rocket:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -164,8 +164,10 @@ Before opening your first pull request, please have a look at this [guide](https
       manually rebase `main` into the issue branch and retrigger a merge
       attempt.
    3. If there are CI errors the author of the pull request has to check if
-      they are caused by its changes and address them. If they are flaky tests
-      a merge can be retried with a comment with the content `bors retry`.
+      they are caused by its changes and address them. If they are flaky tests, please
+      have a look at this [guide](docs/ci.md#determine-flakiness) on how to handle them.
+      Once the CI errors are resolved, a merge can be retried with a comment with
+      the content `bors retry`.
```
Comment (remcowesterhoud) [inline CONTRIBUTING.md:170]: I like the reference 👍

**Review by remcowesterhoud — APPROVED** [review]
Thanks @koevskinikola 🥇 

bors merge

**Comment by korthout** [comment]
Only finding this now, but thanks for this @koevskinikola! 🚀

**Pull request merged**